db/models.py

```diff
@@ -31,22 +33,81 @@
     rows = models.IntegerField()
     seats_in_row = models.IntegerField()
 
-    @property
-    def capacity(self) -> int:
-        return self.rows * self.seats_in_row
-
     def __str__(self) -> str:
         return self.name
 
 
 class MovieSession(models.Model):
+    movie = models.ForeignKey(Movie, on_delete=models.CASCADE)
     show_time = models.DateTimeField()
-    cinema_hall = models.ForeignKey(
-        to=CinemaHall, on_delete=models.CASCADE, related_name="movie_sessions"
-    )
-    movie = models.ForeignKey(
-        to=Movie, on_delete=models.CASCADE, related_name="movie_sessions"
-    )
+    cinema_hall = models.ForeignKey(CinemaHall, on_delete=models.CASCADE)
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE,
+        related_name="orders"
+    )
+
+    def __str__(self) -> str:
+        return self.created_at.strftime("%Y-%m-%d %H:%M:%S")
+
+    class Meta:
+        ordering = ["-created_at"]
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey(MovieSession, on_delete=models.CASCADE)
+    order = models.ForeignKey(Order, on_delete=models.CASCADE)
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["row", "seat", "movie_session"],
+                name="unique_movie_session"
+            )
+        ]
+
+    def __str__(self) -> str:
+        return (
+            f"{self.movie_session.movie.title} "
+            f"{self.movie_session.show_time.strftime('%Y-%m-%d %H:%M:%S')} "
+            f"(row: {self.row}, seat: {self.seat})"
+        )
+
+    def clean(self) -> None:
+        if not 1 <= self.row <= self.movie_session.cinema_hall.rows:
+            raise ValidationError(
+                {
+                    "row": "row number must be in available range: (1, rows): "
+                    f"(1, {self.movie_session.cinema_hall.rows})"
+                }
+            )
+        if not 1 <= self.seat <= self.movie_session.cinema_hall.seats_in_row:
+            raise ValidationError(
+                {
+                    "seat": "seat number must be in available range: "
+                    "(1, seats_in_row): "
+                    f"(1, {self.movie_session.cinema_hall.seats_in_row})"
+                }
+            )
+
+    def save(self, *args, **kwargs) -> None:
+        self.full_clean()
+        super().save(*args, **kwargs)
+
+
+class User(AbstractUser):
+    groups = models.ManyToManyField(
```

Comment:
Code Style issue: It's recommended to use `get_user_model()` instead of `AbstractUser` as a best practice. This way, if the user model changes in the future, you won't need to change your code.